pinot-common/src/main/java/org/apache/pinot/common/utils/UploadedRealtimeSegmentName.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+
+/**
+ * Class to represent segment names like: uploaded_{tableName}_{partitionId}_{sequenceId}_{creationTime}
```

Comment:
this comment might be missed: https://github.com/apache/pinot/pull/13107#discussion_r1619430793 

basically, I think it can be more flexible to use an optional suffix and not enforce the value of the suffix, e.g. `{tableName}_{partitionId}_{sequenceId}_{creationTime}_{optional suffix}` as users can use suffix value to differentiate segments uploaded in different ways (e.g. via minion tasks)

Reply:
The reason to enforce prefix was to quickly identify an uploaded segment. With optional suffix we would have to fall back to segment status in ZK metadta to know if it is an uploaded segment(do suggest if there is a better way).

It may even be complex to parse the segment name to figure out partitionId and segmentId.

Reply:
The final semgent name convention has been decided as `uploaded__{tableName}__{partitionId}__{sequenceId}__{creationTime}__{optionalSuffix}`
This ensures, similar naming as LLCSegmentName, removes need to do ZK lookup in various critical paths, uploaded prefix enforces the type of segment and an optional suffix can be used to encode additional info, e.g. source of segment(spark, flink, minion, etc)